### PR TITLE
 v2

### PR DESCRIPTION
**Pull request by legordian**
Add executables to remove and add integral matrices to fit result files.

### CODE REVIEW

File: src/compareFitResultsIntegrals.cc [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+#include "ampIntegralMatrix.h"
+#include "fileUtils.hpp"
+#include "fitResult.h"
+#include "reportingUtilsEnvironment.h"
+
+using namespace std;
+using namespace rpwa;
+
+
+const ampIntegralMatrix* readIntegralMatrix(const string& intFileName, const string& integralTKeyName = "integral") {
```
Comment (suhlatwork) [inline src/compareFitResultsIntegrals.cc]: This function should be removed from this source file, as it is not used.

File: src/complexMatrix.cc [hunk 2]
```diff
@@ -40,6 +40,59 @@ complexMatrix::~complexMatrix() {
 }
 
 
+bool complexMatrix::operator==(const complexMatrix& rhs) const
+{
+	if(this->nCols() != rhs.nCols() or this->nRows() != rhs.nRows()) {
+		return false;
+	}
+	for(unsigned int i = 0; i < this->nCols(); ++i) {
+		for(unsigned int j = 0; j < this->nCols(); ++j) {
```
Comment (suhlatwork) [inline src/complexMatrix.cc]: This should probably read "j < this->nRows()".

File: src/complexMatrix.cc [hunk 3]
```diff
@@ -40,6 +40,59 @@ complexMatrix::~complexMatrix() {
...
+{
+	bool success = true;
+	if(this->nCols() != rhs.nCols() or this->nRows() != rhs.nRows()) {
+		if(verbose) {
+			printDebug << "dimension mismatch." << endl;
+		}
+		return false;
+	}
+	for(unsigned int i = 0; i < this->nCols(); ++i) {
+		for(unsigned int j = 0; j < this->nCols(); ++j) {
```
Comment (suhlatwork) [inline src/complexMatrix.cc]: This should probably read "j < this->nRows()".

File: src/stripIntegralsFromFitResult.cc [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+		outResult->fill(nmbEvents,
+		                normNmbEvents,
+		                massBinCenter,
+		                logLikelihood,
+		                rank,
+		                prodAmps,
+		                prodAmpNames,
+		                stripCovarianceMatrix ? TMatrixT<double>(0, 0) : inResult->fitParCovMatrix(),
+		                fitParCovMatrixIndices,
+		                normIntegral,  // contains the sqrt of the integral matrix diagonal elements!!!
```
Comment (suhlatwork) [inline src/stripIntegralsFromFitResult.cc]: the comment belongs to phaseSpaceIntegral

File: src/addIntegralsToFitResult.cc [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+		outResult->fill(nmbEvents,
+		                normNmbEvents,
+		                massBinCenter,
+		                logLikelihood,
+		                rank,
+		                prodAmps,
+		                prodAmpNames,
+		                fitParCovMatrix,
+		                fitParCovMatrixIndices,
+		                normIntegral,  // contains the sqrt of the integral matrix diagonal elements!!!
```
Comment (suhlatwork) [inline src/addIntegralsToFitResult.cc]: the comment belongs to phaseSpaceIntegral

**Comment by legordian** [comment]
All problems pointed out by Sebastian Uhl (thank you for that!) have been fixed in commits a196585 and 52d933f.

**Pull request merged**